src/solvers/spqr.jl

```diff
@@ -132,7 +132,7 @@ struct QRSparseQ{Tv<:CHOLMOD.VTypes,Ti<:Integer} <: LinearAlgebra.AbstractQ{Tv}
     τ::Vector{Tv}
     n::Int # Number of columns in original matrix
 end
-
+Base.print_array(io::IO, Q::QRSparseQ) = Base.print_array(io, Q.factors)
```

Comment:
This should perhaps be
```suggestion
Base.print_array(io::IO, Q::QRSparseQ) = Base.print_array(io, sparse(Q))
```
or something like this (see the corresponding issue) which demonstrates the required effort.

Reply:
```
using LinearAlgebra, SparseArrays, Random, BenchmarkTools
Random.seed!(131);
A = sparse(I + sprandn(100, 100, 1/100));
q = qr(A);
@benchmark display($q.Q)
BenchmarkTools.Trial: 52 samples with 1 evaluation.
 Range (min … max):  14.399 ms …    1.835 s  ┊ GC (min … max): 0.00% …  0.00%
 Time  (median):     19.071 ms               ┊ GC (median):    6.26%
 Time  (mean ± σ):   98.233 ms ± 336.622 ms  ┊ GC (mean ± σ):  2.24% ± 10.47%

  █                                                             
  █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▄▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▄ ▁
  14.4 ms       Histogram: log(frequency) by time        1.5 s <

 Memory estimate: 22.70 MiB, allocs estimate: 352056.

Base.print_array(io::IO, Q::SparseArrays.SPQR.QRSparseQ) = Base.print_array(io, sparse(Q))
@benchmark display($q.Q)

BenchmarkTools.Trial: 14 samples with 1 evaluation.
 Range (min … max):  312.287 ms … 445.235 ms  ┊ GC (min … max): 4.86% … 8.65%
 Time  (median):     355.080 ms               ┊ GC (median):    7.42%
 Time  (mean ± σ):   371.133 ms ±  47.507 ms  ┊ GC (mean ± σ):  7.66% ± 1.32%

  ▁ ▁  ▁   ▁  ▁▁   ▁   ▁            ▁     ▁ ▁             ▁   █  
  █▁█▁▁█▁▁▁█▁▁██▁▁▁█▁▁▁█▁▁▁▁▁▁▁▁▁▁▁▁█▁▁▁▁▁█▁█▁▁▁▁▁▁▁▁▁▁▁▁▁█▁▁▁█ ▁
  312 ms           Histogram: frequency by time          445 ms <

 Memory estimate: 763.71 MiB, allocs estimate: 32155.
```

is the performance loss acceptable?

Reply:
after looking at the code it seem that the get index is really not very efficient and the call to sparse gets redirected to the abstract implementation which means that there is a vector being created for every value look up. 

i'll try making the conversion to sparse faster by re-using the workspace.

Reply:
> i'll try making the conversion to sparse faster.

Good luck! We have a

```julia
Matrix{T}(Q::QRSparseQ) where {T} = lmul!(Q, Matrix{T}(I, size(Q, 1), min(size(Q, 1), Q.n)))
```

method (and corresponding `[l/r]mul!` methods), which suggests that you need some multiplication code to obtain any matrix representation, be it sparse or dense. For printing some `QRSparseQ` object, I guess it's actually better to use that dense matrix representation, because it comes out naturally and I'm not sure how sparse the Q matrix generically is. For printing some `QRSparse` object, I wonder if we want to skip printing the Q factor as a matrix, because, well, it kind of actually isn't.